File: python/n594_FindLHS.py
# https://leetcode-cn.com/problems/longest-harmonious-subsequence/
class Solution:
    def findLHS(self, nums) -> int:
        if nums == []: return 0
        count = {}
        for i in nums:
            if i not in count.keys():
                count[i] = 1
            else:
                count[i] += 1
        largest = 0
        key_list = list(count.keys())
        key_list.sort()
        for i in range(1, len(key_list)):
            if key_list[i] - key_list[i-1] == 1:
                largest = max(largest, count[key_list[i]]+count[key_list[i-1]])
        return largest 


if __name__ == "__main__":
    pass         
           

File: cpp/n594_FindLHS.cpp
// https://leetcode-cn.com/problems/longest-harmonious-subsequence/
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
using namespace std;

class Solution {
public:
    int findLHS(vector<int>& nums) {
        int size = nums.size();
        if (size == 0) return 0;
        int i = 0, j = 0, largest = 0;
        sort(nums.begin(), nums.end());
        while (i < size){
            while (i < size && nums[i] - nums[j] <= 1 ) i += 1;
            if (nums[i-1] - nums[j] == 1){ 
                largest = max(largest, i - j);
                while (i < size && nums[i] - nums[j] > 1) j += 1;
            }
            else{
                j = i;
                i += 1;
            }          
        }
        return largest;
    }
};

int main(){
    return 0;
}